components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -1,52 +1,249 @@
+          lw_flux_up, lw_flux_dn
+        );
+    }
+
+    void RRTMGPRadiation::finalize_impl  () {
+        rrtmgp::rrtmgp_finalize();
     }
-    void RRTMGPRadiation::run_impl      (const Real dt) {}
-    void RRTMGPRadiation::finalize_impl  () {}
 
     // Register all required and computed field in the field repository
     void RRTMGPRadiation::register_fields(FieldRepository<Real>& field_repo) const {
         for (auto& fid: m_required_fields) { field_repo.register_field(fid); }
         for (auto& fid: m_computed_fields) { field_repo.register_field(fid); }
     }
 
+    // Private function to check that fields are not padded
+    void RRTMGPRadiation::require_unpadded(const Field<const Real>& f) {
+        const auto& fid = f.get_header().get_identifier();
+        const auto& layout = fid.get_layout();
+        auto v = f.get_view();
+        EKAT_REQUIRE_MSG (
+            layout.size() == v.size(), 
+            "ERROR: field " << fid.name() << " was padded (to allow packing), but currently RRTMGP does not work with padded views."
+        );
+    }
+
     void RRTMGPRadiation::set_required_field_impl(const Field<const Real>& f) {
-        /* The following copied from the P3 code */
-        //const auto& name = f.get_header().get_identifier().name();
-        //m_rrtmgp_fields_in.emplace(name,f);
-        //m_rrtmgp_host_views_in[name] = Kokkos::create_mirror_view(f.get_view());
-        //m_raw_ptrs_in[name] = m_rrtmgp_host_views_in[name].data();
+        const auto& name = f.get_header().get_identifier().name();
+        m_rrtmgp_fields_in.emplace(name,f);
+        m_rrtmgp_host_views_in[name] = Kokkos::create_mirror_view(f.get_view());
+        m_raw_ptrs_in[name] = m_rrtmgp_host_views_in[name].data();
 
         // Add myself as customer to the field
-        //add_me_as_customer(f);
+        add_me_as_customer(f);
+
+        // Check to make sure that fields are not padded because we are not 
+        // sure how to handle that with RRTMGP yet
+        require_unpadded(f);
     }
 
     void RRTMGPRadiation::set_computed_field_impl(const Field<      Real>& f) {
-        /* The following copied from the P3 code */
-        //const auto& name = f.get_header().get_identifier().name();
-        //m_rrtmgp_fields_out.emplace(name,f);
-        //m_rrtmgp_host_views_out[name] = Kokkos::create_mirror_view(f.get_view());
-        //m_raw_ptrs_out[name] = m_rrtmgp_host_views_out[name].data();
+        const auto& name = f.get_header().get_identifier().name();
+        m_rrtmgp_fields_out.emplace(name,f);
+        m_rrtmgp_host_views_out[name] = Kokkos::create_mirror_view(f.get_view());
+        m_raw_ptrs_out[name] = m_rrtmgp_host_views_out[name].data();
 
         // Add myself as provider for the field
-        //add_me_as_provider(f);
+        add_me_as_provider(f);
+
+        // Check to make sure that fields are not padded because we are not 
+        // sure how to handle that with RRTMGP yet
+        require_unpadded(f);
```

Comment:
If this goes in as is we will want to prioritize getting RAD ready for padding since P3 already uses it and I will guess that the SHOC interface will too.  I'm in favor of getting this in sooner than later and putting in a patch for padding when we decide to hook RAD up to the rest of the atm. processes.  If that route is possible.

Reply:
We could probably make a git-issue for this too.  Maybe combined with the git-issue I suggested above, we could have a "remaining RAD tasks" issue with a couple of check-boxes.  This issue would be higher priority, but I think we should work on getting this PR merged first and then tackle this when we begin coupling RAD to the rest of the processes.

Reply:
Thanks @AaronDonahue. I just created an entirely new issue. Contrary to the implication from this huge PR, I prefer to break these things into smaller chunks, and this seems like a well-defined unit of work to tackle in a single PR.